ITISFoundation/osparc-simcore
# End2End system-testing

## What do these changes do?
Adds an End2End system-testing step to the Continuous Integration process.

For this, two main libraries were added:
- [Puppeteer](https://github.com/GoogleChrome/puppeteer) is a Node library that provides a way to control a headless Chrome browser.
- [Jest](https://github.com/facebook/jest) is a testing framework that provides:
  - Testing structure
  - Assertion functions
  - Test progress and results
  - Code Coverage

**How to test**:
The testing is done automatically as an extra step in the CI.

However, if you want to run the tests in your machine:
- go to ops/travis/e2e-testing
- npm install --save
- npm test

If you want to SEE live what the tests are doing, in jest-puppeteer.config.js change "demo" variable to to true before running them.

**Bonus**:
- Sort studies by lastChangeDate
- Study text filter added

![image](https://user-images.githubusercontent.com/33152403/64766259-07650a00-d546-11e9-8e0c-201232d590a2.png)

## Related issue number
child of #634


## Checklist
